tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -3046,4 +3046,52 @@ public static function data_alphabet_by_characters_uppercase() {
 			yield strtoupper( $data[0] ) => array( strtoupper( $data[0] ) );
 		}
 	}
+
+	/**
+	 * Test that script tags are parsed correctly.
+	 *
+	 * Script tag parsing is very complicated, see the following resources for more details:
+	 *
+	 * - https://html.spec.whatwg.org/multipage/parsing.html#script-data-state
+	 * - https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements
+	 *
+	 * @ticket 63738
+	 *
+	 * @dataProvider data_script_tag
+	 */
+	public function test_script_tag_parsing( string $input, bool $closes ) {
+		$processor = new WP_HTML_Tag_Processor( $input );
+
+		if ( $closes ) {
+			$this->assertTrue( $processor->next_token(), 'Expected to find complete script tag.' );
+			$this->assertSame( 'SCRIPT', $processor->get_tag() );
+			return;
+		}
+
+		$this->assertFalse( $processor->next_token(), 'Expected to fail next_token().' );
+		$this->assertTrue( $processor->paused_at_incomplete_token(), 'Expected an incomplete SCRIPT tag token.' );
+	}
```

Comment:
we also have `data_next_tag_ignores_script_tag_contents` in line 1932 which has similar tests. I think this is fine here, though perhaps if we keep it separate we could fit it in neatly up there with the other tests covering SCRIPT tags. just an idea; not important